### ISSUE TITLE
Catch-at-age by month for SRG meeting

### ISSUE THREAD

**Issue by cgrandin** [cgrandin/hake-assessment#266]
I know this is a late request, but the AP are hoping to get catch-at-age by month for all 5 fleets. The Canadian AP in particular is a bit troubled about the catch of 2 year-olds in the US and wonders what the monthly catch was like. Is it possible for you guys to get those values in time? I can do Canada's if we're going to do it.

**Comment by taylori** [comment]
I will work on this for the US fleets. I had meant to do this anyway bit it slipped my mind with so much else to do. I'm assuming we're just talking about 2016 in which case we might be able to fit it into a table in the final assessment document. For now I will just try to get it into a CSV file and we can decide where to go from there.

**Comment by cgrandin** [comment]
Thanks Ian, I'll do the same for Canada. Only 2016 would be ok for this year I think, but what they really want is all years.

**Comment by taylori** [comment]
It might not be that much harder to do it for more years, so I'll try to be generalized and if we choose to add a table to the document we can discuss whether to subset or not.

**Comment by cgrandin** [comment]
Are you going to weight the proportions by sample weight? Or just provide raw proportions from the database?

**Comment by taylori** [comment]
Good question.
For the comps that are in Tables 6-10, Aaron and I were just re-running code that Allan developed without taking the time to understand the details. It looks like his function includes a BY_MONTH option that was not fully implemented. I will take a look at what would be required to get that working to produce weighted proportions, but don't yet have a good sense of what will be involved. I suspect that the weighting among sectors to produce the coastwide comps is much more important that the weighting among tows within any sector. At least in the U.S., I think the tows are often pretty similar in size and the sampling protocols very consistent across tows, so a raw proportion would be close enough and much quicker to calculate.

**Comment by aaronmberger** [comment]
Agree that weighting among sectors will dominate.  When combining comps across fleets, the comps are weighted by biomass at age (total catch adjusted by proportions times weight at age), noting that weight at age does not vary by fleet (one set of values per year)

**Comment by cgrandin** [comment]
The way I do it is to weight by sample weight. If there is no sample weight recorded, sum the weights of the individuals in the sample and use that as the sample weight. If there are individual weights missing in the sample (like in our freezer trawler fleet which has no individual weights), use the L/W relationship of all data to calculate the missing individual weights, then sum those and use them as the sample weight. I'm trying to get that working on a by-month basis but if it's too much coding I'll just add raw proportions.

**Comment by taylori** [comment]
After further investigation, I found that the code for the U.S. At-Sea sectors (Mothership and Catcher-Processor) are already set up to produce monthly summaries of age comps in numbers weighted by sample weight for the years 2008-2016. However, the code for the U.S. Shoreside sector was written to be used for any species of groundfish and is more complicated and doesn't have a monthly option, so I'm less sure of getting it sorted out in time.

**Comment by cgrandin** [comment]
In that case, let's try to do it for 2018.

**Comment by taylori** [comment]
Sounds good.
I can just make a verbal statement about this during the meeting: age-6 fish dominated the catch in the Spring and age-2 were the majority in the Fall. The same pattern occurred in 2012 when the 2010 cohort was age-2. I can provide more detail if requested but this doesn't seem like something we need to get into the document this year.

**Comment by cgrandin** [comment]
I think that will be fine this year - leaving open as I've changed this issue to a 2018 milestone.

**Comment by taylori** [comment]
CSV files with monthly age comps for all sectors re now in Google Drive data folder, but any additional figure or table with information on monthly patterns seems like a lower priority and will depend on having time available once other stuff is complete.
